### ISSUE TITLE
Solution of HH depends on dt

### ISSUE THREAD

**Issue by coschroeder** [jaxleyverse/jaxley#149]
I played around with dt of the stimulus and realized that the number of spikes in the standard HH model depends on the stepsize. 
Code to reproduce:

```python
from neurax.channels import HHChannel

# Number of segments per branch.
nseg_per_branch = 1
comp = nx.Compartment()
branch = nx.Branch([comp for _ in range(nseg_per_branch)])

# point neuron:
cell = nx.Cell([branch for _ in range(1)], parents=jnp.asarray([-1]))

cell.insert(HH())

# Stimulus.
i_delay = 20.0  # ms
i_amp = 3.7  # nA 0.08
i_dur = 20.0  # ms

# Duration and step size.
dt = 0.0025  # ms # Changing dt here, changes the HH solution
t_max = 60.0  # ms

time_vec: jnp.Array = jnp.arange(0.0, t_max+dt, dt)
stim: jnp.Array = jx.step_current(i_delay, i_dur, i_amp, dt, t_max)

cell.branch(0).comp(0).stimulate(stim)
# Solve HH
s = nx.integrate(cell)
```

Am I doing something wrong here?
Or is there a bug in the solver?

**Comment by michaeldeistler** [comment]
To fix this, we could do:
```python
stim, t = jx.step_current(i_delay, i_dur, i_amp, dt, t_max)
cell.branch(0).comp(0).stimulate(stim, t)
```

This has the advantage that `stim` and `t` can both be arrays, which makes handling this very intuitive.
